release/models/platform/openconfig-platform-types.yang

```diff
@@ -299,6 +305,14 @@ module openconfig-platform-types {
       "Linecard component, typically inserted into a chassis slot";
   }
 
+  identity MODULE {
```

Comment:
(insert cliché, naming is the hardest problem...)

I wonder whether we should specifically call this `LINECARD_DAUGHTERCARD` or similar? The reason I suggest this is that I can see some implementation just creating `MODULE` -> `MODULE` -> `MODULE`... which seems like it erodes a lot of the value :-)

Reply:
yep - completely open to suggestions here as I pondered the same for something to generically represent this type of FRU.  How about `LINECARD_MODULE` to at least reel this back to representing an FRU that is a child of `LINECARD` so that `MODULE` doesn't get abused as you mention?

Reply:
I'm happy with that naming.

Reply:
Addressed in https://github.com/openconfig/public/pull/612/commits/6969c7b2bdcee07709cef2bd9873d3ba8eab5fea

Reply:
I like the `LINECARD_MODULE` name better.   Note, there isn't anything preventing arbitrary nesting of these things.  Should we be even more explicit in the description to say the  `LINECARD_MODULE` must have a parent component of type `LINECARD`?

Reply:
@dplore all component types have this issue and an implementation could in theory create a parent/child relationship in any fashion they wish.  Do we want to dictate these types of rules in the descriptions here or have a reference document that outlines suggested and common parent/child relationships? (there will still be variations of this depending on the hardware layouts of platforms)

Reply:
For the `LINECARD_MODULE` leaf, I think it makes sense to add description requiring a parent of type `LINECARD` in this PR.  I think it would be nice to have a reference document recommending a pattern to follow for component trees, but that can be a separate PR if you are motivated to create one.

Reply:
Thx @dplore - addressed in https://github.com/openconfig/public/pull/612/commits/ead3a53b53d30b897333387f39638ff66b71053e